[hotfix] Refactor codes with latest API changes

alnzng opened the pull request:
<!--
* Thank you very much for contributing to Flink Agents.
* Please add the relevant components in the PR title. E.g., [api], [runtime], [java], [python], [hotfix], etc.
-->

<!-- Please link the PR to the relevant issue(s). Hotfix doesn't need this. -->
Linked issue: #xxx

### Purpose of change

<!-- What is the purpose of this change? -->
Refactor codes with latest API changes 

### Tests

<!-- How is this change verified? -->
Run `rag_agent_example.py` locally

### API

<!-- Does this change touches any public APIs? -->
n/a

### Documentation

<!-- Should this change be covered by the user documentation?-->
n/a

xintongsong commented:
Actually, I've notice that the rag agent is under `examples/`, not `examples/quickstart/`.

I think we should have both quick-start examples which should be the simplest examples to get start with, and also other examples showing how to use the core features or dealing with typical scenarios. That's why we have made `quickstart/` a sub-directory of `examples/`. You may also noticed that we have moved other "examples" from `examples/` to `e2e_tests/`, because those are too detailed and more like test cases to prove that an certain features work.

I think the rag agent might be more suitable for the normal examples. Users probably won't need it in the first step, but would look for it when they need to work on some rag-like stuff. That's why I haven't point it out when reviewing the previous PR.

WDYT?

alnzng commented:
Good suggestion @xintongsong ! I agree that current RAG implementation is too advanced for quickstart. Perhaps once we have a built-in RAG agent (similar to ReAct), we can add a simpler example to `examples/quickstart`.

I've moved this example to `examples/rag/` and refactored it to be more focused and user-friendly. The test data setup logic has been extracted to a separate` knowledge_base_setup.py` file, making the main example clearer and more concise for learning how to build RAG agents.

Please take a look, thanks.

xintongsong reviewed (approved):
LGTM
